src/app/template-library/template-library.service.ts

```diff
@@ -0,0 +1,144 @@
+import {Injectable} from '@angular/core';
+import {map} from 'rxjs/operators';
+import {environment} from '@environments/environment';
+import {DemoService} from '@services/demo.service';
+import {RequestService} from '../shared/request/request.service';
+import {Observable} from 'rxjs/Observable';
+
+export interface Template {
+  uuid: string;
+  name: string;
+  description?: string;
+  leadImageUrl?: string;
+  leadVideoUrl?: string;
+  institutionUUID: string;
+  isPublic?: boolean;
+  type?: string;
+  attributes: string[];
+  modified?: string;
+}
+
+export interface Category {
+  name: string;
+  type: string;
+  leadImage: string;
+  color: string;
+  size: Size;
+}
+
+export interface CategorisedTemplates {
+  category: Category;
+  templates: Template[];
+}
+
+enum Size {
+  'LARGE', 'SMALL'
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TemplateLibraryService {
+
+  constructor(
+    private request: RequestService,
+    private demo: DemoService,
+  ) { }
+
+  getTemplates(): Observable<Template[]> {
+    if (environment.demo) {
+      return this.demo.getTemplates().pipe(map(this._handleTemplates));
+    }
+    return this.request.graphQLQuery(
```

Comment:
We don't need to query all the fields if we don't need them.
The beauty of GraphQL is that front-end can control what to get from the server.